From 55a7c8f8c3e51f842f5e2120c58c1646d6cfd764 Mon Sep 17 00:00:00 2001
From: MengLi <meng.li@windriver.com>
Date: Tue, 6 Apr 2021 15:08:31 +0800
Subject: [PATCH 03/13] driver: pci: pci-s32gen1: remove the __init macro

There is a section mismatch building warning as below:
WARNING: modpost: vmlinux.o(.text+0x6ddfac): Section
mismatch in reference from the function s32gen1_pcie_probe()
to the function .init.text:s32gen1_add_pcie_port()
Because function s32gen1_add_pcie_port() is invoked in
s32gen1_pcie_probe() that has not the __init macro, remove
__init macro from function s32gen1_add_pcie_port() and
s32gen1_add_pcie_ep() to avoid this building warning.

Upstream-Status: Pending

Signed-off-by: Meng Li <Meng.Li@windriver.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/pci/controller/dwc/pci-s32gen1.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/pci/controller/dwc/pci-s32gen1.c b/drivers/pci/controller/dwc/pci-s32gen1.c
index 9ba726e5c9ea..eec9f873c2a6 100644
--- a/drivers/pci/controller/dwc/pci-s32gen1.c
+++ b/drivers/pci/controller/dwc/pci-s32gen1.c
@@ -770,7 +770,7 @@ static int s32gen1_pcie_config_irq(int *irq_id, char *irq_name,
 	return 0;
 }
 
-static int __init s32gen1_add_pcie_port(struct pcie_port *pp,
+static int s32gen1_add_pcie_port(struct pcie_port *pp,
 			struct platform_device *pdev)
 {
 	int ret;
@@ -822,7 +822,7 @@ static struct dw_pcie_ep_ops pcie_ep_ops = {
 	.raise_irq = s32gen1_pcie_ep_raise_irq,
 };
 
-static int __init s32gen1_add_pcie_ep(struct s32gen1_pcie *s32_pp,
+static int s32gen1_add_pcie_ep(struct s32gen1_pcie *s32_pp,
 				     struct platform_device *pdev)
 {
 	int ret;
-- 
2.25.1

